src/components/NotificationProvider/NotificationProvider.tsx

```diff
@@ -0,0 +1,103 @@
+import React, {
+  createContext,
+  FC,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import {
+  NotificationType,
+  NotificationHelper,
+  QueuedNotification,
+  NotifyProviderProps,
+} from "./types";
+import { useLocation } from "react-router-dom";
+import isEqual from "lodash/isEqual";
+import { info, failure, success, queue } from "./messageBuilder";
+import Notification, { DefaultTitles } from "../Notification/Notification";
+
+const NotifyContext = createContext<NotificationHelper>({
+  notification: null,
+  clear: () => undefined,
+  failure: () => undefined,
+  success: () => undefined,
+  info: () => undefined,
+  queue: () => undefined,
+});
+
+export const NotificationProvider: FC<NotifyProviderProps> = ({ children }) => {
+  const [notification, setNotification] = useState<NotificationType | null>(
+    null
+  );
+  const { state, pathname } = useLocation() as QueuedNotification;
+
+  const clear = () => notification !== null && setNotification(null);
+
+  const setDeduplicated = (value: NotificationType) => {
+    if (!isEqual(value, notification)) {
+      setNotification(value);
+    }
+    return value;
+  };
+
+  useEffect(() => {
+    if (state?.queuedNotification) {
+      setDeduplicated(state.queuedNotification);
+      window.history.replaceState({}, "");
+    } else {
+      clear();
+    }
+  }, [state, pathname]);
```

Comment:
Apparently this makes the linter a bit unhappy:
```
  51:6  warning  React Hook useEffect has missing dependencies: 'clear' and 'setDeduplicated'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
```
It's just a warning, but maybe we can fix it?

Reply:
Including them would run the hook too often so we'll have to accept the warning.